File: kde/kalzium/kalzium.orig/src/kalzium-23.08.4/src/solver/main.ml
(*
    SPDX-FileCopyrightText: 2004 Thomas Nagy <tnagy2^8@yahoo.fr>

    SPDX-License-Identifier: GPL-2.0-or-later
*)

open Printf;;
open Calc;;


let myeq1 = " a CH3(CH2)3COOH + b O2 -> c H2O + d CO2";;
let myeq2 = " a CH3COOH + b O2 -> c H2O + d CO2";;
let myeq3 = " a CH3(CH2)3COOH + b O2 -> invalid equation ";;
let myeq4 = "2CH4+aC2H6+cO2->eH2O+5CO2";;
let myeq5 = "a NaNO2 + b NH3 -> b NaN3 + e NaNO3";;
let myeq6 = "a HNO2 + b NH3 -> b HN3 + e HNO3";;
let myeq7 = "a H3O[+] + dCO2 + eO2 -> fH2O + dCO3[2+]";;

(* Printf.printf "%s \n" (solve_equation myeq1);;
Printf.printf "%s \n" (solve_equation myeq2);;
Printf.printf "%s \n" (solve_equation myeq3);; *)
(*
Printf.printf "%s \n" (solve_equation myeq5);;
Printf.printf "%s \n" (solve_equation myeq6);;
Printf.printf "%s \n" (solve_equation myeq6);;

Printf.printf "%s \n" (solve_equation myeq6);;*)
Printf.printf "%s \n" (solve_equation myeq7);;

File: kde/kalzium/kalzium.orig/src/kalzium-23.08.4/src/solver/main.c
/*
    SPDX-FileCopyrightText: 2004 Thomas Nagy <tnagy2^8@yahoo.fr>

    SPDX-License-Identifier: GPL-2.0-or-later
*/

#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv)
{
    caml_startup(argv);

    char *eq = " a CH3(CH2)3COOH + b O2 -> c H2O + d CO2";
    char *eq2 = " a CH3(CH2)3COOH + b O2 -> c H2O + d CO";
    char *result = solve_equation(eq);
    char *result2 = solve_equation(eq2);

    printf("solution : %s\n", result);

    printf("solution : %s\n", result2);
    free(result2);

    result2 = solve_equation(eq);
    printf("solution : %s\n", result2);

    free(result);
    free(result2);

    return 0;
}
